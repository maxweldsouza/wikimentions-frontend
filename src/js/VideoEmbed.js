import React from 'react';
import queryString from 'query-string';
import parseUrl from 'url-parse';
import utils from './utils';

const YoutubeEmbed = (
    {
        url,
        embeddable,
        width = 640,
        height = 390,
        autoplay = true
    }
) => {
    let embed;
    const parsed = parseUrl(url);
    if (utils.isYoutubeUrl(url)) {
        const queryObject = queryString.parse(parsed.query);
        const query = autoplay ? '?autoplay=1' : '';
        embed =
            <iframe
                id="ytplayer"
                type="text/html"
                width={width}
                height={height}
                src={`https://www.youtube.com/embed/${queryObject.v}${query}`}
                frameBorder={0}
            />
        ;
    } else {
        return null;
    }
    if (!embeddable) {
        return null;
    }
    return (
        <div className="flex-video widescreen">
            {embed}
        </div>
    );
};

export default YoutubeEmbed;
