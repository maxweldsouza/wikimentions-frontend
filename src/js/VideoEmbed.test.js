import React from 'react';
import renderer from 'react-test-renderer';
import VideoEmbed from './VideoEmbed';

test('VideoEmbed renders correctly', () => {
    const tree = renderer
        .create(<VideoEmbed url='https://www.youtube.com/watch?v=BBCFQtDLPA0' />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
