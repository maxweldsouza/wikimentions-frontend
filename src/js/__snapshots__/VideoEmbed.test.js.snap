// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoEmbed renders correctly 1`] = `null`;
